components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/Nimbus.kt

```diff
@@ -444,52 +481,108 @@ open class Nimbus(
     internal fun fetchExperimentsOnThisThread() = withCatchAll {
         try {
             nimbusClient.fetchExperiments()
-            observer?.onExperimentsFetched()
+            updateObserver {
+                it.onExperimentsFetched()
+            }
         } catch (e: NimbusException.RequestException) {
             errorReporter("Error fetching experiments from endpoint", e)
         } catch (e: NimbusException.ResponseException) {
             errorReporter("Error fetching experiments from endpoint", e)
         }
     }
 
-    override fun applyPendingExperiments() {
-        dbScope.launch {
-            applyPendingExperimentsOnThisThread()
+    private fun updateObserver(updater: (NimbusInterface.Observer) -> Unit) {
+        val observer = observer ?: return
+        val scope = updateScope ?: return updater(observer)
```

Comment:
```suggestion
        val scope = updateScope ?: return
```

This is odd. We're returning the `updater` function itself back to the caller, but the function returns a `Unit` so we're not really returning anything.

Maybe keep this consistent with the line above?

Reply:
I've changed this to a more verbose implementation: I wanted to run the updater even if there wasn't an `updateScope` to run it on a different scope.